src/scheduler.rs

```diff
@@ -194,31 +202,30 @@ struct InnerSchedulerState {
     paused: bool,
 }
 
+/// Guard to make sure the IO Scheduler is resumed.
+///
+/// Returned by [`SchedulerState::pause`].  To resume the IO scheduler simply drop this
+/// guard.
 #[derive(Debug)]
 pub(crate) struct IoPausedGuard {
-    context: Context,
-    done: bool,
+    sender: Option<oneshot::Sender<()>>,
 }
 
 impl IoPausedGuard {
-    pub(crate) async fn resume(&mut self) {
-        self.done = true;
-        let mut inner = self.context.scheduler.inner.write().await;
-        inner.paused = false;
-        if inner.started && inner.scheduler.is_none() {
-            SchedulerState::do_start(inner, self.context.clone()).await;
-        }
+    /// Resume the scheduler explicitly.
+    ///
+    /// Does nothing special, simply drops this guard.
+    pub(crate) fn resume(self) {
+        drop(self)
```

Comment:
arguably we could remove this method.  i couldn't make up my mind so feel free to have an opinion.